onnxruntime/core/providers/dml/DmlExecutionProvider/src/MLOperatorAuthorImpl.cpp

```diff
@@ -982,6 +982,85 @@ namespace Windows::AI::MachineLearning::Adapter
         m_abiExecutionObject.CopyTo(executionInterface);
     }
 
+    uint32_t STDMETHODCALLTYPE OpKernelInfoWrapper::GetUtf8NameSizeInBytes() const noexcept
+    {
+        // Include null terminator.
+        const auto& name = m_impl->node().Name(); 
+        return name.empty() ? 0 : name.size() + 1;
```

Comment:
Not sure if it matters (without seeing where/how it's used), but if `Name()` is an `std::string`, then even an empty std string has a null terminator, making it 1 byte long. So the extra logic could be reduced to just `return name.size() + 1;`, and a memcpy of `c_str()` will do the right thing.

Reply:
The idea is that a return value of 0 indicates an empty string, so the caller doesn't need to dynamically allocate memory and both calling GetName(). I suppose this is logically equivalent to returning 1 if the underlying string is empty, since a size of 1 with null terminator implies an empty string.

I'll update to what you suggest.